### PR TITLE
RN TP 3

### PR DESCRIPTION
**Pull request by Preeticp**
This PR adds release notes for Pipelines TP 1.3 
It has been ack'd by SME/QE
Aligned team: Dev Tools
Applicable to 4.7

### CODE REVIEW

File: modules/op-release-notes-1-3.adoc [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+= Release notes for {pipelines-title} Technology Preview 1.3
+
+[id="new-features-1-3_{context}"]
+== New features
+{pipelines-title} Technology Preview (TP) 1.3 is now available on {product-title} 4.7. {pipelines-title} TP 1.3 is updated to support:
+
+* Tekton Pipelines 0.19.0
+* Tekton `tkn` CLI 0.15.0
+* Tekton Triggers 0.10.2
+* ClusterTasks based on Tekton Catalog 0.16.3
```
Comment (savitaashture) [inline modules/op-release-notes-1-3.adoc:15]:
```suggestion
* ClusterTasks based on Tekton Catalog 0.19.0
```

File: modules/op-release-notes-1-3.adoc [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+{pipelines-title} Technology Preview (TP) 1.3 is now available on {product-title} 4.7. {pipelines-title} TP 1.3 is updated to support:
+
+* Tekton Pipelines 0.19.0
+* Tekton `tkn` CLI 0.15.0
+* Tekton Triggers 0.10.2
+* ClusterTasks based on Tekton Catalog 0.19.0
+* IBM Power Systems on {product-title} 4.7
+* IBM Z and LinuxONE on {product-title} 4.7
+
+In addition to the fixes and stability improvements, here is a highlight of what’s new in OpenShift Pipelines 1.3.
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:19]: improvements, the following sections highlight what is new in in OpenShift Pipelines 1.3.

File: modules/op-release-notes-1-3.adoc [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+=== Pipelines
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:21]: This heading should have its own unique ID; same with Pipelines CLI, Triggers, and subsequent headings.

File: modules/op-release-notes-1-3.adoc [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+* Variable expansion is now added in the `Task` CRD for the following fields:
+`spec.steps[].imagePullPolicy` and `spec.sidecar[].imagePullPolicy`.
+
+* You can disable the built-in credential mechanism in Tekton by setting the `disable-creds-init` feature-flag to `true`.
+
+* Resolved `When` expressions are now listed in the `Skipped Tasks` and the `Task Runs` sections in the `Status` field of the `PipelineRun` configuration.
+
+* The `git init` command can now clone recursive submodules.
+
+* A `Task` CR author can now specify a timeout for a Step in the `Task` spec.
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:36]: s/Step/step

File: modules/op-release-notes-1-3.adoc [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+* You can now base the entry point image on `distroless/static:nonroot` and give it a mode to copy itself to the destination, without relying on the `cp` command being present in the base image.
+
+* You can now use the configuration flag `require-git-ssh-secret-known-hosts` to disallow omitting known hosts in the Git SSH secret. When the flag value is set to `true`, you must include the `known_host` field in the Git SSH secret. The default value for the flag is `false`.
+
+* The concept of optional workspaces is now introduced. A task or pipeline may declare a workspace optional and conditionally change their behavior based on its presence. A task run or pipeline run may also omit that workspace, thereby modifying the task or pipeline behavior. Note that the default task run workspaces are not added in place of an omitted optional workspace.
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:42]: s/may/might
s/Note that the default/The default

File: modules/op-release-notes-1-3.adoc [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+* Tekton's credentials initialization now detects an SSH credential that is used with a non-SSH URL, and vice versa in Git pipeline resources, and logs a warning in the step containers.
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:44]: Credentials initialization in Tekton now

File: modules/op-release-notes-1-3.adoc [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+* The task run controller emits a warning event if the affinity specified by the pod template is overwritten by the affinity assistant.
+
+* The task run reconciler now records metrics for cloud events that are emitted once a taskrun is completed. This includes retries.
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:48]: s/taskrun/task run

File: modules/op-release-notes-1-3.adoc [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+=== Triggers
+
+* You can now specify your resource information in the `EventListener` template.
+
+* It is now mandatory for `EventListener` service accounts to have the `list` and `watch` verbs, in addition to the `get` verb for all the triggers resources. This enables you to use `Listers` to fetch data from `EventListeners`, `Triggers`, `TriggerBindings`, `TriggerTemplates`, and `ClusterTriggerBindings` resources. You can use this feature to create a `Sink` object rather than specifying multiple informers, and directly make calls to the API server.
+
+* A new Interceptor interface is added to support immutable input event bodies. Interceptors can now add data or fields to a new `extensions` field, and cannot modify the input bodies making them immutable. The CEL interceptor uses this new Interceptor interface.
+
+* A `namespaceSelector` field is added to the `EventListener` resource. Use it to specify the namespaces from where the `EventListener` resource can fetch the `Trigger` object for processing events.
+Note that to use the `namespaceSelector` field, the service account for the `EventListener` must have a cluster role.
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:77]: s/Note that to use/To use

File: modules/op-release-notes-1-3.adoc [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+
+* The escaping parameters behavior in the `TriggerTemplates` resource by replacing `"` with `\"` is now removed.
+
+* A new `resources` field, supporting Kubernetes resources, is introduced as part of the `EventListener` spec.
+
+* A new functionality for the CEL interceptor, with support for upper and lower-casing of ASCII strings, is added.
+
+* You can embed `TriggerBinding` resources by using the `name` and `value` fields in a trigger, or an event listener.
+
+* The `PodSecurityPolicy` configuration is updated to run in restricted environments. It ensures that containers must run as non-root. In addition, the role-based access control for utilizing the pod security policy is moved from cluster-scoped to namespace-scoped. This ensures that the triggers cannot utilize other pod security policies that are unrelated to a namespace.
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:89]: s/utilizing/using
s/utilize/use

File: modules/op-release-notes-1-3.adoc [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+
+* Support for embedded trigger templates is now added. You can either use the `name` field to refer to an embedded template or embed the template inside the `spec` field.
+
+
+[id="deprecated-features-1-3_{context}"]
+== Deprecated features
+
+* Pipeline templates that use PipelineResources CRDs are now deprecated and will be removed in a future release.
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:97]: `PipelineResources` CRDs

File: modules/op-release-notes-1-3.adoc [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+
+* The `template.name` field is deprecated in favor of the `template.ref` field and will be removed in a future release.
+
+* The `-c` shorthand for the `--check` command has been removed. In addition, global `tkn` flags are added to the `version` command.
+
+
+[id="known-issues-1-3_{context}"]
+== Known issues
+
+* CEL overlays add fields to a new top level `extensions` function, instead of modifying the incoming event body. `TriggerBinding` resources can access values within this new `extensions` function using the `$(extensions.<key>)` syntax. Update your binding to use `$(extensions.<key>)` instead of `$(body.<overlay-key>)` .
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:107]: top-level

Comment (ahardin-rh) [reply modules/op-release-notes-1-3.adoc:107]: What does CEL stand for?

File: modules/op-release-notes-1-3.adoc [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+
+* The escaping parameters behavior by replacing `"` with `\"` is now removed. If you need to retain the old escaping parameters behavior add the `tekton.dev/old-escape-quotes: true"` annotation to your `TriggerTemplate` specification.
+
+* You can embed `TriggerBinding` resources by using the `name` and `value` fields inside a trigger or an event listener. However, you cannot specify both `name` and `ref` fields for a single binding. Use the `ref` field to refer to a `TriggerBinding` resource and the `name` field for embedded bindings.
+
+* An interceptor cannot attempt to reference a `Secret` outside the namespace of an `EventListener`. You must include secrets in the namespace of the `EventListener`.
+
+* In Triggers 0.9.0 and later, if a body or header based `TriggerBinding` parameter is missing or malformed in an event payload, the default values are used instead of displaying an error.
+
+* `Tasks` and `Pipelines` created with `WhenExpressions` using Tekton Pipelines 0.16.x need to be reapplied to fix their JSON annotations.
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:117]: s/need to be/must be

File: modules/op-release-notes-1-3.adoc [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+
+* Extensions added by a CEL Interceptor are passed on to webhook interceptors by adding the `Extensions` field within the event body.
+
+* The activity timeout for log readers is now configurable using the `LogOptions` field. However, the default behavior of timeout in 10 seconds is retained.
+
+* The `log` command ignores the `--follow` flag when a task run or pipeline run is complete, and reads available logs instead of live logs.
+
+* References to the following Tekton resources: `EventListener`, `TriggerBinding`, `ClusterTriggerBinding`, `Condition`, and `TriggerTemplate` are now standardized and made consistent across all user-facing messages in `tkn` commands.
+
+* Previously, if you started a canceled task run or pipeline run with the `--use-taskrun <canceled-task-run-name>`, `--use-pipelinerun <canceled-pipeline-run-name>` or `--last` flags, the new run would be canceled. This bug has now been fixed.
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:135]: This bug is now fixed.

File: modules/op-release-notes-1-3.adoc [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+
+* The `tkn pr desc` command  is now enhanced to ensure that it does not fail in case of pipeline runs with conditions.
+
+* When you delete a task run using `tkn tr delete` with the `--task` option, and a cluster task exists with the same name, the task runs for the cluster task also gets deleted. As a workaround, filter the task runs by using the `TaskRefKind` field.
+
+* The `tkn triggertemplate describe` command would display only part of the `apiVersion` in the output. For example, only `triggers.tekton.dev` was displayed instead of `triggers.tekton.dev/v1alpha1`. This bug has been fixed.
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:141]: This bug is now fixed.

File: modules/op-release-notes-1-3.adoc [hunk 15]
```diff
@@ -0,0 +1,153 @@
...
+
+* The webhook, under certain conditions, would fail to acquire a lease and not function correctly. This bug has been fixed.
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:143]: This bug is now fixed.

File: modules/op-release-notes-1-3.adoc [hunk 16]
```diff
@@ -0,0 +1,153 @@
...
+
+* Pipelines with `When` expressions created in v0.16.3 can now be run in v0.17.1 and later. After an upgrade, you do not need to reapply pipeline definitions created in previous versions because both the uppercase and lowercase first letters for the annotations are now supported.
+
+* By default, the `leader-election-ha` is now enabled for high availability. When the controller flag `disable-ha` is set to `true`, it disables high availability support.
+
+* Issues with duplicate cloud events are now fixed. Cloud events are now sent only when a condition changes the state, reason, or message.
+
+* When a service account name is missing from a `PipelineRun` or `TaskRun` spec the controller uses the service account name from the `config-defaults` config map. If it is missing in the `config-defaults` config map too, the controller now sets it to `default` in the spec.
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:151]: spec, the controller

If the service account name is also missing in the the `config-defaults` config map

File: modules/op-release-notes-1-3.adoc [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+
+* Validation for compatibility with the affinity assistant is now supported when the same persistent volume claim is used for multiple workspaces, but with different sub paths.
```
Comment (ahardin-rh) [inline modules/op-release-notes-1-3.adoc:153]: s/sub paths/subpaths

**Comment by ahardin-rh** [comment]
@Preeticp Just some minor style edits from me. :+1:

**Comment by Preeticp** [comment]
Thank you @ahardin-rh I have fixed all comments and squashed the commits. Please merge if this looks good.

**Review by jeana-redhat — APPROVED** [review]
approving exception

**Pull request merged**